### ISSUE TITLE
feature(client): create as context manager

### ISSUE THREAD

**Issue by aarnphm** [bentoml/BentoML#3288]
### Feature request

Client could become a sync/async context manager:
```python
with Client.from_url("localhost:3000") as client:
    client.predict()
```
async version
```python
async with Client.from_url("...") as client:
   await client.async_predict()
```

### Motivation

implement a `__enter__`, `__exit__`, `__aenter__`, and `__aexit__` for client.

**Comment by y1450** [comment]
I am new the repository, looking for my first issue to contribute. Is it okay if I can work on it.

**Comment by sauyon** [comment]
Sure, this should hopefully be pretty simple!

**Comment by sauyon** [comment]
Feel free to ping me on slack if you need any guidance :)

**Comment by aarnphm** [comment]
This has now been included in 1.0.13
